sqlalchemy_bigquery/base.py

```diff
@@ -686,6 +771,147 @@ def visit_drop_table_comment(self, drop):
+            )
+
+    def _process_time_partitioning(
+        self, table: Table, time_partitioning: TimePartitioning
+    ):
+        """
+        Generates a SQL 'PARTITION BY' clause for partitioning a table by a date or timestamp.
+
+        Args:
+        - table (Table): The SQLAlchemy table object representing the BigQuery table to be partitioned.
+        - time_partitioning (TimePartitioning): The time partitioning details,
+            including the field to be used for partitioning.
+
+        Returns:
+        - str: A SQL 'PARTITION BY' clause that uses either TIMESTAMP_TRUNC or DATE_TRUNC to
+        partition data on the specified field.
+
+        Example:
+        - Given a table with a TIMESTAMP type column 'event_timestamp' and setting
+        'time_partitioning.field' to 'event_timestamp', the function returns
+        "PARTITION BY TIMESTAMP_TRUNC(event_timestamp, DAY)".
+        """
+        field = "_PARTITIONDATE"
+        trunc_fn = "DATE_TRUNC"
+
+        if time_partitioning.field is not None:
+            field = time_partitioning.field
+            if isinstance(
+                table.columns[time_partitioning.field].type,
+                sqlalchemy.sql.sqltypes.TIMESTAMP,
+            ):
+                trunc_fn = "TIMESTAMP_TRUNC"
+
+        return f"PARTITION BY {trunc_fn}({field}, {time_partitioning.type_})"
+
+    def _process_range_partitioning(
+        self, table: Table, range_partitioning: RangePartitioning
+    ):
+        """
+        Generates a SQL 'PARTITION BY' clause for partitioning a table by a range of integers.
+
+        Args:
+        - table (Table): The SQLAlchemy table object representing the BigQuery table to be partitioned.
+        - range_partitioning (RangePartitioning): The RangePartitioning object containing the
+        partitioning field, range start, range end, and interval.
+
+        Returns:
+        - str: A SQL string for range partitioning using RANGE_BUCKET and GENERATE_ARRAY functions.
+
+        Raises:
+        - ValueError: If the partitioning field is not defined, not an integer type, or if the range
+        start/end values are not integers.
+
+        Example:
+            "PARTITION BY RANGE_BUCKET(zipcode, GENERATE_ARRAY(0, 100000, 10))"
+        """
+        if range_partitioning.field is None:
+            raise ValueError("bigquery_range_partitioning expects field to be defined")
+
+        if not isinstance(
+            table.columns[range_partitioning.field].type,
+            sqlalchemy.sql.sqltypes.INT,
+        ):
+            raise ValueError(
+                "bigquery_range_partitioning expects field data type to be INTEGER"
+            )
+
+        range_ = range_partitioning.range_
+
+        if not isinstance(range_.start, int):
+            raise ValueError(
+                "bigquery_range_partitioning expects range_.start to be an int,"
+                f" provided {repr(range_.start)}"
+            )
```

Comment:
```suggestion
            raise TypeError(
                "bigquery_range_partitioning expects range_.start to be an INTEGER,"
                f" provided {repr(range_.start)}"
            )
```

Also correct the type check and the term INTEGER for `range_.end` in the next expression, below.

I have not checked yet to see if any tests look for a specific kind of Error in this location. IF yes, those tests will need an update, as well.

Reply:
You're right about the error type. I'll get that fixed.

Regarding the error message, I choose to differentiate INTEGER (the column data type) and int (the Python variable data type). However I'm not aware of the standard way for this kind of error message in Python, so let me know if this should still be adjusted.

Reply:
I am fine with using int, now that I have a better understanding of what we are referring to here.
Thanks for explaining and for your patience.